### PR TITLE
i18n(fr): Add `google-firebase.mdx` translation

### PR DESCRIPTION
**Pull request by djibh**
#### Description (required)

Creating a new doc for Google Firebase integration.

### CODE REVIEW

File: src/content/docs/fr/guides/backend/google-firebase.mdx [hunk 1]
```diff
@@ -0,0 +1,930 @@
...
+    <button type="submit">Login</button>
+  </form>
+</Layout>
+```
+
+`src/pages/signin.astro` utilise l'application serveur Firebase pour vérifier le cookie de session de l'utilisateur. Si l'utilisateur est authentifié, la page redirigera l'utilisateur vers la page `/dashboard`.
+
+La page d'exemple ci-dessous contient un formulaire qui enverra une requête « POST » au endpoint « /api/auth/signin » avec le token d'identification généré par l'application client Firebase.
+
+Le endpoint vérifiera le token d'identification et créera un nouveau cookie de session pour l'utilisateur. Ensuite, le endpoint redirigera l'utilisateur vers la page « /dashboard ».
```
Comment (McFlyPartages) [inline src/content/docs/fr/guides/backend/google-firebase.mdx:348]:
```suggestion
Le point de terminaison vérifiera le token d'identification et créera un nouveau cookie de session pour l'utilisateur. Ensuite, le point de terminaison redirigera l'utilisateur vers la page « /dashboard ».
```

File: src/content/docs/fr/guides/backend/google-firebase.mdx [hunk 2]
```diff
@@ -0,0 +1,930 @@
...
+    if (response.redirected) {
+      window.location.assign(response.url);
+    }
+  });
+</script>
+```
+
+`src/pages/dashboard.astro` vérifiera le cookie de session de l'utilisateur à l'aide de l'application serveur Firebase. Si l'utilisateur n'est pas authentifié, la page redirigera l'utilisateur vers la page `/signin`.
+
+La page d'exemple ci-dessous affiche le nom de l'utilisateur et un bouton pour se déconnecter. Cliquer sur le bouton enverra une requête « GET » au endpoint « /api/auth/signout ».
```
Comment (McFlyPartages) [inline src/content/docs/fr/guides/backend/google-firebase.mdx:422]:
```suggestion
La page d'exemple ci-dessous affiche le nom de l'utilisateur et un bouton pour se déconnecter. Cliquer sur le bouton enverra une requête `GET` au point de terminaison `/api/auth/signout`.
```

File: src/content/docs/fr/guides/backend/google-firebase.mdx [hunk 3]
```diff
@@ -0,0 +1,930 @@
...
+Créez les pages qui utiliseront les endpoints Firestore:
+
+- `src/pages/add.astro` - contiendra un formulaire permettant d'ajouter un nouvel ami ("friend").
+- `src/pages/edit/[id].astro` - contiendra un formulaire permettant de modifier un ami et un bouton permettant de supprimer un ami.
+- `src/pages/friend/[id].astro` - contiendra les informations d'un ami.
+- `src/pages/dashboard.astro` - affichera une liste d'amis.
+
+#### Ajouter une nouvelle entrée
+
+L'exemple `src/pages/add.astro` ci-dessous inclut un formulaire qui enverra une requête `POST` au endpoint `/api/friends`. Ce endpoint créera un nouvel ami en utilisant les données du formulaire, puis redirigera l'utilisateur vers la page `/dashboard`.
```
Comment (McFlyPartages) [inline src/content/docs/fr/guides/backend/google-firebase.mdx:714]:
```suggestion
L'exemple `src/pages/add.astro` ci-dessous inclut un formulaire qui enverra une requête `POST` au point de terminaison `/api/friends`. Ce point de terminaison créera un nouvel ami en utilisant les données du formulaire, puis redirigera l'utilisateur vers la page `/dashboard`.
```

File: src/content/docs/fr/guides/backend/google-firebase.mdx [hunk 4]
```diff
@@ -0,0 +1,930 @@
...
+    <label for="isBestFriend">Is best friend?</label>
+    <input type="checkbox" id="isBestFriend" name="isBestFriend" />
+    <button type="submit">Add friend</button>
+  </form>
+</Layout>
+```
+
+#### Modifier ou supprimer une entrée
+
+`src/pages/edit/[id].astro` contiendra un formulaire pour modifier les données d'un ami et un bouton pour supprimer un ami. Lors de la soumission, cette page enverra une requête `POST` au endpoint `/api/friends/:id` pour mettre à jour les données d'un ami.
```
Comment (McFlyPartages) [inline src/content/docs/fr/guides/backend/google-firebase.mdx:737]:
```suggestion
`src/pages/edit/[id].astro` contiendra un formulaire pour modifier les données d'un ami et un bouton pour supprimer un ami. Lors de la soumission, cette page enverra une requête `POST` au point de terminaison `/api/friends/:id` pour mettre à jour les données d'un ami.
```

File: src/content/docs/fr/guides/backend/google-firebase.mdx [hunk 5]
```diff
@@ -0,0 +1,930 @@
...
+
+Si l'utilisateur clique sur le bouton Supprimer, cette page enverra une requête « DELETE » au endpoint « /api/friends/:id » pour supprimer un ami.
```
Comment (McFlyPartages) [inline src/content/docs/fr/guides/backend/google-firebase.mdx:739]:
```suggestion
Si l'utilisateur clique sur le bouton Supprimer, cette page enverra une requête `DELETE` au point de terminaison `/api/friends/:id` pour supprimer un ami.
```

File: src/content/docs/fr/guides/backend/google-firebase.mdx [hunk 6]
```diff
@@ -0,0 +1,930 @@
...
+    <input
+      type="checkbox"
+      id="isBestFriend"
+      name="isBestFriend"
+      checked={friend.isBestFriend}
+    />
+    <button type="submit">Edit friend</button>
+  </form>
+  <button type="button" id="delete-document">Delete</button>
+</Layout>
```
Comment (McFlyPartages) [inline src/content/docs/fr/guides/backend/google-firebase.mdx:788]:
```suggestion
  <h1>Editer {friend.name}</h1>
  <p>Ici on peut supprimer ou modifier les données d'un ami.</p>
  <form method="post" action={`/api/friends/${id}`}>
    <label for="name">Nom</label>
    <input type="text" id="name" name="name" value={friend.name} />
    <label for="age">Age</label>
    <input type="number" id="age" name="age" value={friend.age} />
    <label for="isBestFriend">C'est mon meilleur ami ?</label>
    <input
      type="checkbox"
      id="isBestFriend"
      name="isBestFriend"
      checked={friend.isBestFriend}
    />
    <button type="submit">Editer l'ami</button>
  </form>
  <button type="button" id="delete-document">Supprimer</button>
</Layout>
```

File: src/content/docs/fr/guides/backend/google-firebase.mdx [hunk 7]
```diff
@@ -0,0 +1,930 @@
...
+  return Astro.redirect("/404");
+}
+
+const friend = friendSnapshot.data() as Friend;
+---
+
+<Layout title={friend.name}>
+  <h1>{friend.name}</h1>
+  <p>Age: {friend.age}</p>
+  <p>Is best friend: {friend.isBestFriend ? "Yes" : "No"}</p>
```
Comment (McFlyPartages) [inline src/content/docs/fr/guides/backend/google-firebase.mdx:841]:
```suggestion
  <p>C'est mon meilleur ami: {friend.isBestFriend ? "Yes" : "No"}</p>
```

File: src/content/docs/fr/guides/backend/google-firebase.mdx [hunk 8]
```diff
@@ -0,0 +1,930 @@
...
+const friendsRef = db.collection("friends");
+const friendsSnapshot = await friendsRef.get();
+const friends = friendsSnapshot.docs.map((doc) => ({
+  id: doc.id,
+  ...doc.data(),
+})) as Friend[];
+---
+
+<Layout title="My friends">
+  <h1>Friends</h1>
```
Comment (McFlyPartages) [inline src/content/docs/fr/guides/backend/google-firebase.mdx:872]:
```suggestion
  <h1>Amis</h1>
```

File: src/content/docs/fr/guides/backend/google-firebase.mdx [hunk 9]
```diff
@@ -0,0 +1,930 @@
...
+  <ul>
+    {
+      friends.map((friend) => (
+        <li>
+          <a href={`/friend/${friend.id}`}>{friend.name}</a>
+          <span>({friend.age})</span>
+          <strong>{friend.isBestFriend ? "Bestie" : "Friend"}</strong>
+          <a href={`/edit/${friend.id}`}>Edit</a>
```
Comment (McFlyPartages) [inline src/content/docs/fr/guides/backend/google-firebase.mdx:880]:
```suggestion
          <a href={`/edit/${friend.id}`}>Editer</a>
```

File: src/content/docs/fr/guides/backend/google-firebase.mdx [hunk 10]
```diff
@@ -0,0 +1,930 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (McFlyPartages) [inline src/content/docs/fr/guides/backend/google-firebase.mdx:930]:
```suggestion
```

**Review by McFlyPartages — COMMENTED** [review]
Salut et merci pour ce super travail.

Voici quelques modifications.

Pour information, dans les traductions, on ne change pas la mise en page (cf. `un mot` par "un mot").
Il y a aussi un petit guide (pas évident  à trouver, j'avoue) sur les traductions utilisées pour essayer d'uniformiser les traductions. https://github.com/withastro/docs/tree/main/src/i18n/fr

Mais merci encore à toi et super travail.

File: src/content/docs/fr/guides/backend/google-firebase.mdx [hunk 11]
```diff
@@ -0,0 +1,930 @@
...
+Lorsque vous cliquez dessus, le bouton de connexion Google ouvrira une fenêtre contextuelle (popup) pour vous connecter avec Google. Une fois que l'utilisateur s'est connecté, il enverra une requête « POST » au endpoint « /api/auth/signin » avec le token d'identification généré par le fournisseur OAuth.
+
+Le endpoint vérifiera le token d'identification et créera un nouveau cookie de session pour l'utilisateur. Ensuite, le endpoint redirigera l'utilisateur vers la page « /dashboard ».
+
+## Connexion à Firestore Database
+
+### Prérequis
+
+- Un projet Astro initialisé avec Firebase comme décrit dans la section [Initialisation de Firebase dans Astro](#Initialisation-de-Firebase-dans-Astro).
```
Comment (sarah11918) [inline src/content/docs/fr/guides/backend/google-firebase.mdx:563]:
```suggestion
- Un projet Astro initialisé avec Firebase comme décrit dans la section [Initialisation de Firebase dans Astro](#initialisation-de-firebase-dans-astro).
```

**Comment by sarah11918** [comment]
@djibh Merci pour une grande contribution! C'est fantastique pour un premier PR!

Just so you know how we manage our PRs here, when someone makes a code suggestion, there will be a button for you to "accept" their new code. This is the easiest way to update your submission with corrections or ideas from other people! Only you and Astro maintainers are allowed to change your code, so normally it should be you that comes back and presses that button (over and over again...) to accept other people's suggestions. The other translators cannot change your code for you.

Since I can also do it, I have done so for all the suggestions left here, to save you the effort. In future, please do come back and "accept" suggestions that you agree with! Or, leave comments and discuss if there is any doubt or disagreement.

Then, in future, when everyone agrees that all points are resolved, an Astro maintainer will look for agreement (we use "LGTM!" for "Looks Good to Me!") and know that all translators have agreed on the translation. Then, we will merge the PR.

Since this PR has been thoroughly reviewed, I will merge it now as soon as I check the deploy preview for any obvious formatting/errors in the build. If there are still issues to address, anyone can then make another PR to correct something that missed this review! But, it is better to get PRs merged quickly (before the page changes AGAIN...) and the content live on the site helping people. Future PRs for small issues are always welcome, and ensures that content doesn't sit here too long and become outdated.

Merci encore, bienvenue à Team Docs, et j'attends votre prochaine contribution! 🥳

File: src/content/docs/fr/guides/backend/google-firebase.mdx [hunk 12]
```diff
@@ -0,0 +1,915 @@
...
+type: backend
+service: Firebase
+stub: false
+i18nReady: true
+---
+import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro'
+import FileTree from '~/components/FileTree.astro'
+
+
+[Firebase](https://firebase.google.com/)est une plate-forme de développement d'applications qui fournit une base de données NoSQL, une authentification, des souscriptions en temps réel, des fonctions et un stockage.
```
Comment (sarah11918) [inline src/content/docs/fr/guides/backend/google-firebase.mdx:13]:
```suggestion
[Firebase](https://firebase.google.com/) est une plate-forme de développement d'applications qui fournit une base de données NoSQL, une authentification, des souscriptions en temps réel, des fonctions et un stockage.
```

**Pull request merged**